homeassistant/components/template/binary_sensor.py

```diff
@@ -133,15 +133,33 @@ def __init__(
         self._template = value_template
         self._state = None
         self._delay_cancel = None
-        self._delay_on = delay_on
-        self._delay_off = delay_off
+        self._delay_on = None
+        self._delay_on_raw = delay_on_raw
+        self._delay_off = None
+        self._delay_off_raw = delay_off_raw
         self._unique_id = unique_id
 
     async def async_added_to_hass(self):
         """Register callbacks."""
 
         self.add_template_attribute("_state", self._template, None, self._update_state)
 
+        try:
```

Comment:
Actually we should probably check to see if `self._delay_on_raw is not None` before doing the try block.

Reply:
I moved the if condition from the `except` block to wrap around the entire `try`. The overall behaviour of the code remains the same but it saves a few function calls 😉